### PR TITLE
update raft to v1.3.8

### PR DESCRIPTION
**Pull request by dhiaayachi**
This update raft to [v1.3.7](https://github.com/hashicorp/raft/releases/tag/v1.3.7)
In this raft release `HeartbeatTimeout` and `ElectionTimeout` became reloadable config, so Consul needed to be tweaked to only reload those configs only when the given values are within the acceptable range.
Note that this also introduce https://github.com/hashicorp/raft/pull/477

### CODE REVIEW

File: .changelog/12844.txt
Comment (eculver) [inline .changelog/12844.txt:2]: should this be `v1.3.7`?

Comment (eculver) [inline .changelog/12844.txt:2]:
```suggestion
raft: upgrade to v1.3.7 which fixes a bug where non cluster member can still be able to participate in an election.
```

**Comment by eculver** [comment]
I'm curious, what are the implications of making these fields "reloadable"? Does this mean

**Comment by eculver** [comment]
Also, it looks like `v1.3.7` includes https://github.com/hashicorp/raft/pull/477 which may be worth describing in the PR description for posterity.

File: .changelog/12844.txt
Comment (dhiaayachi) [reply .changelog/12844.txt:2]: good catch, I made a typo writing it

**Review by mkeeler — APPROVED** [review]
Approved assuming the changelog is updated as Evan pointed out.

**Review by mkeeler — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/650650.

**Pull request merged**
